components/layout_2020/replaced.rs

```diff
@@ -406,15 +409,15 @@ impl ReplacedContent {
                         // does not itself depend on the replaced element's width,
                         // then the used value of 'width' is calculated from the constraint
                         // equation used for block-level, non-replaced elements in normal flow.”
-                        _ => default_object_size().inline,
+                        _ => default_object_size().inline.into(),
                     };
                 let block_size = if let Some(block) = intrinsic_size.block {
                     block
                 } else if let Some(i_over_b) = intrinsic_ratio {
                     // “used width” in CSS 2 is what we just computed above
-                    inline_size / i_over_b
+                    inline_size / i_over_b as i32
```

Comment:
Seems better to use `scale_by(1.0 / i_over_b)`, e.g.

```rs
let inline_size = Au::from_px(1);
let i_over_b = 31.0;
println!("{:?}", inline_size / i_over_b as i32);        // 0.016666666666666666px
println!("{:?}", inline_size.scale_by(1.0 / i_over_b)); // 0.03333333333333333px
```

And in math `1 / 31` is closer to the latter.

Reply:
Thanks for the tip. `i_over_b as i32` did made me uncomfortable. Updated in [28cfcbe](https://github.com/servo/servo/pull/31178/commits/28cfcbe7048a152c952043bf73054126c2b38169)